### PR TITLE
Labels ilias

### PR DESCRIPTION
**Pull request by nop33**
Instead of commenting all these improvements/bugs to @LeeAlephium 's PR, I went ahead and implemented the things I noticed in my functional tests. 😇  I would appreciate some quite tests from your side since you have different transaction data than I do. I might have missed something.

https://user-images.githubusercontent.com/1579899/191739167-886b3bb3-8c1b-48c7-af54-b10ff0283f3a.mp4

After recording the video I noticed that the pending txs do not get displayed at all (no data). This was fixed as well.
image

### CODE REVIEW

**Comment by LeeAlephium** [comment]
Quick comment

> [Bring back main address indication](https://github.com/alephium/desktop-wallet/pull/364/commits/b9f4f43fcdf2b3bdce9ffcb341ac9c638e3f9eca)

@mvaivre asked to remove

File: src/components/Ellipsed.tsx [hunk 1]
```diff
@@ -57,10 +57,10 @@ const Ellipsed = ({ text, className }: EllipsedProps) => {
 
   const handleResize = createHandleResize(el, charWidth, text, setText)
 
-  useLayoutEffect(() => {
+  useEffect(() => {
```
Comment (LeeAlephium) [inline src/components/Ellipsed.tsx:60]: Why this change?

File: src/modals/AddressOptionsModal.tsx [hunk 2]
```diff
@@ -74,7 +72,7 @@ const AddressOptionsModal = ({ address, onClose }: AddressOptionsModal) => {
       : t`To remove this address from being the main address, you must set another one as main first.`
 
   return (
-    <div ref={divRef} tabIndex={0}>
```
Comment (LeeAlephium) [inline src/modals/AddressOptionsModal.tsx:77]: Could you show me the broken scrolling?

This div is needed for accessibility purposes

File: src/components/AddressBadge.tsx [hunk 3]
```diff
@@ -34,18 +35,19 @@ type AddressBadgeProps = ComponentPropsWithoutRef<typeof Badge> & {
...
 const AddressBadge = ({ address, showHashWhenNoLabel, className, ...props }: AddressBadgeProps) => {
   const { t } = useTranslation('App')
+  const { isPassphraseUsed } = useGlobalContext()
 
   if (!address) return null
 
   return showHashWhenNoLabel && !address.settings.label ? (
     <Hash className={className}>
-      {address.settings.isMain && '★'}
+      {!isPassphraseUsed && address.settings.isMain && '★'}
```
Comment (LeeAlephium) [inline src/components/AddressBadge.tsx:44]: As mentioned main addresses no longer have a star

File: src/pages/Wallet/AddressesPage.tsx [hunk 4]
```diff
@@ -107,7 +107,7 @@ const AddressesPage = () => {
               </ClipboardButton>
             </TableCell>
             <TableCell role="cell" tabIndex={0}>
-              <AddressBadge address={address} truncate />
+              {address.settings.label ? <AddressBadge address={address} truncate /> : '-'}
```
Comment (LeeAlephium) [inline src/pages/Wallet/AddressesPage.tsx:109]: I would add this logic inside the `AddressBadge`

**Comment by LeeAlephium** [comment]
At a glance it's no longer possible to tell if an address belongs to you or not - the dot was a nice indicator

Example:

File: src/pages/Wallet/AddressesPage.tsx (hunk 4 above)
Comment (LeeAlephium) [reply src/pages/Wallet/AddressesPage.tsx:109]: Hm, it's pretty specific to this context; all good

**Comment by LeeAlephium** [comment]
The non-ellipsed / address-as-label is still shown in address summary cards:

**Comment by LeeAlephium** [comment]
Nice job, just those small things and we'll merge it in :sparkles:

File: src/modals/AddressOptionsModal.tsx (hunk 2 above)
Comment (LeeAlephium) [reply src/modals/AddressOptionsModal.tsx:77]: It's possible it was addressed in https://github.com/alephium/desktop-wallet/pull/353

File: src/components/AddressBadge.tsx (hunk 3 above)
Comment (nop33) [reply src/components/AddressBadge.tsx]: Where was it mentioned, I am sorry if I missed it.

File: src/modals/AddressOptionsModal.tsx (hunk 2 above)
Comment (nop33) [reply src/modals/AddressOptionsModal.tsx:77]: It is shown in the video. The unexpected scrolling happens when opening the address options modal. Removing the div does not affect accessibility, I can still navigate with the Tab key in the modal. Am I missing sth?

File: src/components/AddressBadge.tsx (hunk 3 above)
Comment (LeeAlephium) [reply src/components/AddressBadge.tsx]: https://github.com/alephium/desktop-wallet/pull/364#issuecomment-1254982029

File: src/components/Ellipsed.tsx (hunk 1 above)
Comment (nop33) [reply src/components/Ellipsed.tsx:60]: To solve the issue where the hash of the selected address was not changing in the Send modal (AddressSelect) (demonstrated in the video). I had to put `text` into the depedencies of a `useEffect` and since the body of it was the same as the one in `useLayoutEffect`, I merged them into one `useEffect`. I read the docs for `useLayoutEffect` and it says to always prefer `useEffect` unless you really need to use `useLayoutEffect`. Since everything still works, I went for the simpler solution. Did I miss sth?

Comment (LeeAlephium) [reply src/components/Ellipsed.tsx:60]: `useLayoutEffect` was purposefully used because the measurements needed for the ellipsis must occur after DOM updates - it may appear to work in this instance but the problem could manifest at another time - this happened to me while I was working on it

The dependency is ok :ok_hand:

**Comment by nop33** [comment]
> At a glance it's no longer possible to tell if an address belongs to you or not - the dot was a nice indicator

I am of the opinion that there shouldn't be a distinction between not-labeled addresses and any other external address. If you want a distinction, add a label. I feel like it complicates the UI.

@mvaivre has some ideas on adding a "tag" icon on hover to encourage users to add labels to their own addresses. I propose that we re-iterate on this later, I am personally satisfied with the current state of the UI.

**Comment by nop33** [comment]
> The non-ellipsed / address-as-label is still shown in address summary cards:

That's fine by me, it's on purpose!

**Comment by LeeAlephium** [comment]
Okie dokie :+1: 

I noticed there are several places where address badges are missing the new borders (Addresses screen, address selecting when sending, as examples)

**Comment by nop33** [comment]
> I noticed there are several places where address badges are missing the new borders (Addresses screen, address selecting when sending, as examples)

That's also intentional. I'll leave it up to @mvaivre to fine-tune the rest, for me it looks great as it is now!

File: src/components/AddressBadge.tsx (hunk 3 above)
Comment (nop33) [reply src/components/AddressBadge.tsx]: Now that the "dot" is moved on the right side, I think it's ok to have the star back!

File: src/modals/AddressOptionsModal.tsx (hunk 2 above)
Comment (nop33) [reply src/modals/AddressOptionsModal.tsx:77]: The focus moves on the title of the modal, is that not enough?

Comment (LeeAlephium) [reply src/modals/AddressOptionsModal.tsx:77]: Ah yeah, it is - forgot `CenteredModal` takes care of this now

**Comment by mvaivre** [comment]
Ah, I was going to do some functional tests (and possible tweaks) as soon as I'm done with my current task. Not a big deal, I'll open another PR if necessary.
